cmd/artifact/main.go

```diff
@@ -0,0 +1,341 @@
+		},
+	}
+
+	app.Commands = []cli.Command{
+		{
+			Name:    "download",
+			Aliases: []string{"d"},
+			Usage:   "download an artifact",
+			Flags: []cli.Flag{
+				cli.StringFlag{
+					Name:   "output, o",
+					Usage:  "`FILENAME` to write output to.  If not provided, Standard Output will be assumed",
+					EnvVar: "ARTIFACT_OUTPUT",
+				},
+				cli.BoolFlag{
+					Name:  "latest",
+					Usage: "request artifact from latest run",
+				},
+			},
+			ArgsUsage: "taskId runId name",
+			Action: func(c *cli.Context) error {
+				var err error
+				q, err := queue.New(&tcclient.Credentials{
+					ClientID:    c.GlobalString("client-id"),
+					AccessToken: c.GlobalString("access-token"),
+					Certificate: c.GlobalString("certificate"),
+				})
+				if err != nil {
+					return cli.NewExitError(err, ErrBadUsage)
+				}
+
+				if c.GlobalIsSet("base-url") {
+					q.BaseURL = c.GlobalString("base-url")
+				}
+
+				client := artifact.New(q)
+
+				if c.GlobalBool("allow-insecure-requests") {
+					client.AllowInsecure = true
+				}
+
+				if c.GlobalBool("quiet") {
+					artifact.SetLogOutput(ioutil.Discard)
+				}
+
+				if c.GlobalIsSet("chunk-size") {
+					cz := c.GlobalInt("chunk-size")
+					_, ps := client.GetInternalSizes()
+					err := client.SetInternalSizes(cz*1024, ps)
+					if err != nil {
+						return cli.NewExitError(err, ErrBadUsage)
+					}
+				}
+
+				if !c.IsSet("output") {
+					return cli.NewExitError("must specify output", ErrBadUsage)
+				}
+
+				output, err := os.Create(c.String("output"))
+				if err != nil {
+					return cli.NewExitError(err, ErrInternal)
+				}
+				defer output.Close()
+
+				if c.Bool("latest") {
+					if c.NArg() != 2 {
+						return cli.NewExitError("--latest requires two arguments", ErrBadUsage)
+					}
+					err = client.DownloadLatest(c.Args().Get(0), c.Args().Get(1), output)
+				} else {
+					if c.NArg() != 3 {
+						return cli.NewExitError("three arguments required", 1)
+					}
+					err = client.Download(c.Args().Get(0), c.Args().Get(1), c.Args().Get(2), output)
+
+				}
+
+				switch err {
+				case nil:
+					return nil
+				case artifact.ErrBadOutputWriter:
+					return cli.NewExitError(err, ErrBadUsage)
+				case artifact.ErrCorrupt:
+					return cli.NewExitError(err, ErrCorrupt)
+				default:
+					return cli.NewExitError(err, ErrInternal)
+				}
+			},
+			Category: "Downloading",
+		},
+		{
+			Name:    "upload",
+			Aliases: []string{"u"},
+			Usage:   "upload an artifact",
+			Flags: []cli.Flag{
+				cli.StringFlag{
+					Name:   "input, i",
+					Usage:  "`FILENAME` to read as artifact.  Must be seekable",
+					EnvVar: "ARTIFACT_INPUT",
+				},
+				cli.StringFlag{
+					Name:   "tmp-dir",
+					Usage:  "`DIRECTORY` to write temporary files in",
+					EnvVar: "ARTIFACT_TMPDIR",
+				},
+				cli.BoolFlag{
+					Name:  "gzip",
+					Usage: "serve artifact with gzip content-encoding",
+				},
+				cli.BoolFlag{
+					Name:  "multi-part",
```

Comment:
The word [multipart](https://www.w3.org/Protocols/rfc1341/7_2_Multipart.html) isn't hyphenated so this should be `multipart` rather than `multi-part`.

It is one of those strange cases like backslash vs forward-slash, where one is hyphenated and the other isn't. That's the English language for you! 😄